File: My Leet Code Java Solutions/Solution35.java
/*35. Search Insert Position
    -if target == found
        -return index
    -else
        -return the index where it would be if it were inserted in order
    -O(log n) complexity
*/
class Solution {
    public int searchInsert(int[] nums, int target) {
        for(int i=0; i<nums.length; i++){
            if(target == nums[i]){
                return i;
            }
        }
        int j=0;
        while(target>=nums[j]){
            j++;
            if(j == nums.length){
                break;
            }
        }
        return j;
    }
}

File: MyLeetCodeSolutions/PythonSolutions/Solution35.py
#35. Search Insert Position
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        j = 0
        for i in range(0, len(nums)):
            if target == nums[i]:
                return i
        while target >= nums[j]:
            j += 1
            if j == len(nums):
                break
        return j
